src/amp-story-ad-preview.js

```diff
@@ -129,4 +162,38 @@ export default class AmpStoryAdPreview {
     // b) reject when invalid
     this.writeToIframe_(await this.adIframe_, patch(dirty));
   }
+
+  async setMetaCtaLabel_(dirty) {
+    const {ctaType, ctaUrl} = this.extractMetaCta_(dirty);
+    const storyCta = (await this.storyIframe_).contentDocument.querySelector(
+      '.i-amphtml-story-ad-link'
+    );
+    storyCta.textContent = CTA_TYPES[ctaType];
+    storyCta.setAttribute('href', ctaUrl);
+  }
+
+  extractMetaCta_(dirty) {
+    const head = dirty.substring(0, dirty.lastIndexOf('</head>'));
+    const components = head.split('<');
+    let ctaType = defaultCtaType;
+    let ctaUrl = defaultCtaUrl;
+    for (let component of components) {
+      if (component.includes(`meta name="amp-cta-type"`)) {
```

Comment:
This fails when:
1. `content` is before `name`.
2. Single-quotes `'` or no quotes are used for attribute values.
3. Tag name (`META`) or attributes (`NAME`) are **`UPPERCASE`**.

Let's use a regex to match the `<meta>` tags directly and use a detached node to parse them :)

```js

const metaCtaRe =
  /<meta\s+[^>]*name=['"]?amp-cta-(type|url)['"]?[^>]*>/ig;

class AmpStoryAdPreview {
  constructor(/* ... */) {
    // ...
    this.htmlParserEl_ = null;
  }
  
  // ...
  
  extractMetaCta_(dirty) {
    let ctaType = defaultCtaType;
    let ctaUrl = defaultCtaUrl;
    const matches = dirty.match(metaCtaRe);
    if (matches.length <= 0) {
      return {ctaType, ctaUrl};
    }
    if (!this.htmlParserEl_) {
      this.htmlParserEl_ = this.win.document.createElement('div');
    }
    this.htmlParserEl_.innerHTML = matches.join('\n');
    const metas = this.htmlParserEl_.querySelectorAll('meta');
    this.htmlParserEl_.innerHTML = null;
    for (const meta of metas) {
      const name = meta.getAttribute('name');
      const content = meta.getAttribute
      if (name == 'amp-cta-type') {
        ctaType = content;
      }
      if (name == 'amp-cta-url') {
        ctaUrl = content;
      }
    }
    return {ctaType, ctaUrl};
  }
}
```